pkg/types/validation/installconfig.go

```diff
@@ -336,8 +336,12 @@ func validateNetworking(n *types.Networking, singleNodeOpenShift bool, fldPath *
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("networkType"), n.NetworkType, "networkType Kuryr is not supported on OpenShift later than 4.14"))
 	}
 
-	if singleNodeOpenShift && n.NetworkType == string(operv1.NetworkTypeOpenShiftSDN) {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("networkType"), n.NetworkType, "networkType OpenShiftSDN is currently not supported on Single Node OpenShift"))
+	if n.NetworkType == string(operv1.NetworkTypeOpenShiftSDN) {
+		if singleNodeOpenShift {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("networkType"), n.NetworkType, "networkType OpenShiftSDN is currently not supported on Single Node OpenShift"))
+		} else {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("networkType"), n.NetworkType, "networkType OpenShiftSDN is deprecated, please use OVNKubernetes"))
```

Comment:
There are other `OpenShiftSDN` checks that can go away too. Earlier in this file there's a check to not allow ipv6 when using openshift-sdn, and elsewhere there's a special case to match `networkType: OpenShiftSDN` case-insensitively for backward compatibility with very very very old install-configs.

You should probably just grep for `OpenShiftSDN`... of course, none of those need to be fixed just to remove sdn as an install option, but you'll want to clean them up eventually.